src/common/Cond.h

```diff
@@ -97,6 +97,9 @@ class C_SaferCond : public Context {
   /// Returns rval once the Context is called
   int wait() {
     std::unique_lock l{lock};
+    if (done) {
```

Comment:
i don't think this change is necessary, as a typical `std::condition_variable::wait()` is implemented like
```c++
template<typename Predicate>
void wait(unique_lock<mutex>& lock, Predicate p) {
  while (!p()) {
    wait(lock);
  }
}
```